src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt

```diff
@@ -312,24 +312,20 @@ class ChatListener {
             }
             if (activePet != null) {
                 var rarity = "§f"
-                if (activePet.tier.toString().uppercase() == "UNCOMMON") {
-                    rarity = "§a"
-                } else if (activePet.tier.toString().uppercase() == "RARE") {
-                    rarity = "§9"
-                } else if (activePet.tier.toString().uppercase() == "EPIC") {
-                    rarity = "§5"
-                } else if (activePet.tier.toString().uppercase() == "LEGENDARY") {
-                    rarity = "§6"
-                } else if (activePet.tier.toString().uppercase() == "MYTHIC") {
-                    rarity = "§d"
+                when(activePet.tier.toString().uppercase()) {
+                    ItemRarity.UNCOMMON.rarityName -> rarity = "§a"
+                    ItemRarity.RARE.rarityName -> rarity = "§9"
+                    ItemRarity.EPIC.rarityName -> rarity = "§5"
+                    ItemRarity.LEGENDARY.rarityName -> rarity = "§6"
+                    ItemRarity.MYTHIC.rarityName -> rarity = "§d"
                 }
 
-                component.append(UTextComponent(rarity + activePet.type.toString().lowercase().replace("_", " ")
-                    .replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()}
+                component.append(UTextComponent(rarity + activePet.type.lowercase().replace("_", " ")
+                    .replaceFirstChar { it.uppercase() }
                         + "\n\n")
                     .setHoverText( if(activePet.heldItem == null) {"§cNo Pet Item"} else "§b" +
-                            activePet.heldItem.toString().lowercase().replace("pet_item_", "").replace("_", " ")
-                                .replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()}
+                            activePet.heldItem!!.lowercase().replace("pet_item_", "")
+                                .replace("_", " ").replaceFirstChar { it.uppercase() }
```

Comment:
You can use kotlin's nullable handling to add a default instead of using an if/else